src/main/java/org/takes/facets/fork/MediaType.java

```diff
@@ -108,12 +75,73 @@ public int compareTo(final MediaType type) {
      */
     public boolean matches(final MediaType type) {
         final String star = "*";
-        return (this.high.equals(star)
-            || type.high.equals(star)
-            || this.high.equals(type.high))
-            && (this.low.equals(star)
-            || type.low.equals(star)
-            || this.low.equals(type.low));
+        return (this.highPart().equals(star)
+            || type.highPart().equals(star)
+            || this.highPart().equals(type.highPart()))
+            && (this.lowPart().equals(star)
+            || type.lowPart().equals(star)
+            || this.lowPart().equals(type.lowPart()));
+    }
+
+    /**
+     * Splits the text parts.
+     * @return The parts of the media type.
+     */
+    @Cacheable(forever = true)
+    private String[] split() {
+        return this.text.split(";", 2);
```

Comment:
@dalifreire so how many this method will be called if it was without `@Cacheable`? Moreover, I think that cost of splitting the string with limit is pretty much neglible